compiler/luci/pass/src/ResolveCustomOpMaxPoolWithArgmaxPass.cpp

```diff
@@ -0,0 +1,746 @@
+  auto output_cast = create_cast(concat, loco::DataType::FLOAT32, output_dtype);
+  init_name_and_origin(output_cast, name + "/Cast", origin);
+
+  return output_cast;
+}
+
+bool resolve_max_pool_with_argmax(luci::CircleCustom *cop)
+{
+#define CHECK_OR_FALSE(condition) \
+  if (not(condition))             \
+    return false;
+
+  const std::vector<uint8_t> custom_options = cop->custom_options();
+  auto map = flexbuffers::GetRoot(custom_options).AsMap();
+
+  // Define params
+  // Note: Only `Targmax` equal to DT_INT64 is supported by tflite converter
+  // Note: Only `data_format` equal to "NHWC" is supported by tflite converter
+  // TODO add support of `include_batch_in_index` param
+  auto ksize_param = to_vector<uint32_t>(map["ksize"].AsTypedVector());
+  auto strides_param = to_vector<uint32_t>(map["strides"].AsTypedVector());
+  auto padding_param = map["padding"].As<std::string>();
+
+  // Batch size and depth of ksize more than 1 is not supported.
+  CHECK_OR_FALSE(ksize_param.size() == 4);
+  CHECK_OR_FALSE(ksize_param[0] == 1 && ksize_param[3] == 1);
+
+  CHECK_OR_FALSE(strides_param.size() == 4);
+  CHECK_OR_FALSE(strides_param[0] == 1 && strides_param[3] == 1);
+
+  // define Padding
+  auto padding = string_to_padding(padding_param);
+
+  // define Filter
+  luci::Filter filter;
+  filter.h(ksize_param[1]);
+  filter.w(ksize_param[2]);
+
+  // define Stride
+  luci::Stride stride;
+  stride.h(strides_param[1]);
+  stride.w(strides_param[2]);
+
+  // input node
+  auto const input = loco::must_cast<luci::CircleNode *>(cop->inputs(0));
+  CHECK_OR_FALSE(input->dtype() == loco::DataType::FLOAT32);
+  CHECK_OR_FALSE(input->rank() == 4);
+
+  // TODO support batch size > 1 and `include_batch_in_index` option
+  CHECK_OR_FALSE(input->dim(0).value() == 1);
+
+  // output nodes
+  auto const outputs = loco::succs(cop);
+  CHECK_OR_FALSE(outputs.size() == 2);
+  assert(outputs.size() == cop->numOutputs());
+
+  auto output0 = get_custom_output(cop, 0);
+  auto output1 = get_custom_output(cop, 1);
+
+  // From TF documentation: output of maxpool must has same type as input
+  assert(output0->dtype() == input->dtype());
+  assert(output1->dtype() == loco::DataType::S64 || output1->dtype() == loco::DataType::S32);
+
+  // Create MaxPool
+  auto maxpool = max_pool_branch(padding, stride, filter, cop);
+  auto argmax = argmax_branch(padding, stride, filter, cop);
+
+  // last argmax branch op is cast, it should have dtype initialized
+  assert(argmax->dtype() == output1->dtype());
+
+  // replace old node with new subgraph
+  cop->inputs(0, nullptr);
+  loco::replace(output0).with(maxpool);
+  loco::replace(output1).with(argmax);
+
+  return true;
+}
+
+} // namespace
+
+namespace luci
+{
+
+/**
+ * BEFORE
+ *                 |
+ *            [CircleNode]
+ *                 |
+ *     [CUSTOM(MaxPoolWithArgmax]
+ *         |              |
+ *  [MaxPool output]  [Argmax output]
+ *
+ * AFTER
```

Comment:
Wow..... O.O

Reply:
Yeah, this is the problem with this PR, it is hard to review such complex computations =_=